server/src/immich/api-v1/asset/asset.core.ts

```diff
@@ -52,8 +53,15 @@ export class AssetCore {
       isOffline: dto.isOffline ?? false,
     });
 
+    await this.repository.upsertExif({ assetId: asset.id, fileSizeInByte: file.size });
```

Comment:
Will this break extract metadata missing job, as now all assets will always have a record in the exif table even if that job hasn't run?

Reply:
This is a great question.

Reply:
The filesize should just be in the `asset` table IMO, it's not an exif property.

Reply:
Should implement that functionality in main and then rebase this branch?

Reply:
I think it makes sense to have it in a separate PR, yeah. Changing that feels beyond the scope, especially because it's already kinda big

Reply:
We need to at least ensure the job runs correctly like in my original comment, happy for that to be changed separately

Reply:
@zackpollard Yeah I believe you're right and this would break the missing jobs at the moment. Moving the size to the asset would fix that though, so I feel like that's a good way of solving it

Reply:
Mitigated by changing how we query for "missing" on main.